src/cartridges/int_adyen_SFRA/cartridge/controllers/middlewares/adyen/cancelPartialPaymentOrder.js

```diff
@@ -0,0 +1,40 @@
+const BasketMgr = require('dw/order/BasketMgr');
+const Transaction = require('dw/system/Transaction');
+const AdyenConfigs = require('*/cartridge/scripts/util/adyenConfigs');
+const adyenCheckout = require('*/cartridge/scripts/adyenCheckout');
+const collections = require('*/cartridge/scripts/util/collections');
+
+function cancelPartialPaymentOrder(req, res, next) {
+  try {
+    const request = JSON.parse(req.body);
+    const { splitPaymentsOrder } = request;
+
+    const cancelOrderRequest = {
+      merchantAccount: AdyenConfigs.getAdyenMerchantAccount(),
+      order: splitPaymentsOrder,
+    };
+
+    const response = adyenCheckout.doCancelPartialPaymentOrderCall(
+      cancelOrderRequest,
+    );
+
+    if (response.resultCode && response.resultCode === 'Received') {
+      const currentBasket = BasketMgr.getCurrentBasket();
+      Transaction.wrap(() => {
+        collections.forEach(currentBasket.getPaymentInstruments(), (item) => {
+          if (item.custom.adyenSplitPaymentsOrder) {
+            currentBasket.removePaymentInstrument(item);
+          }
+        });
+      });
+      session.privacy.giftCardResponse = null;
+    }
+
+    res.json(response);
+    return next();
+  } catch (error) {
+    return next();
```

Comment:
if we're not handling any errors we might as well move the return statement to under the try/catch block

Reply:
We could consider logging if that's not already handled by the function calling the service